src/awkward/contents/listoffsetarray.py

```diff
@@ -207,25 +207,26 @@ def to_ListOffsetArray64(self, start_at_zero=False):
     def to_RegularArray(self):
         start, stop = self._offsets[0], self._offsets[self._offsets.length - 1]
         content = self._content._getitem_range(slice(start, stop))
-        size = ak.index.Index64.empty(1, self._backend.index_nplike)
+        _size = ak.index.Index64.empty(1, self._backend.index_nplike)
         assert (
-            size.nplike is self._backend.index_nplike
+            _size.nplike is self._backend.index_nplike
             and self._offsets.nplike is self._backend.index_nplike
         )
         self._handle_error(
             self._backend[
                 "awkward_ListOffsetArray_toRegularArray",
-                size.dtype.type,
+                _size.dtype.type,
                 self._offsets.dtype.type,
             ](
-                size.data,
+                _size.data,
```

Comment:
It's not very common to prepend _local_ variables hidden in a function definition with underscores. Is that just to ensure that this variable is distinct from the one on (green) line 226, also named `size`?

Reply:
Yes. Now that lengths and offsets are fundamentally different types, we want to be very careful not to mix them unknowingly. This is becoming a pattern; wherever a kernel computes a length, we want the ability to realise the unknown-ness in the standard `None | int` type space.